core/src/main/java/edu/stanford/futuredata/macrobase/pipeline/CubePipeline.java

```diff
@@ -167,4 +158,36 @@ private CubeClassifier getClassifier() throws MacrobaseException {
                 throw new MacrobaseException("Bad Classifier Name");
         }
     }
+
+    private Explanation getExplanation(CubeClassifier classifier, DataFrame df) throws Exception {
+        switch (classifierType) {
+            case "meanshift": {
+                APLMeanSummarizer summarizer = new APLMeanSummarizer();
+                summarizer.setCountColumn(countColumn);
+                summarizer.setMeanColumn(meanColumn);
+                summarizer.setStdColumn(stdColumn);
+                summarizer.setAttributes(attributes);
+                summarizer.setMinSupport(minSupport);
+                summarizer.setMinStdDev(minRatioMetric);
+                long startTime = System.currentTimeMillis();
```

Comment:
I think it's better to put running the summarizer, timing it, and getting the results in the main pipeline. The helper functions should just return a "Summarizer" object though that might take a bit of refactoring.

Reply:
Do we want to make a superclass for BatchSummarizer and APLSummarizer? They don't share much functionality at the moment.

Reply:
You can use APLMeanSummarizer and APLOutlierSummarizer which both implement APLSummarizer so we just always use the new code for the cubepipeline